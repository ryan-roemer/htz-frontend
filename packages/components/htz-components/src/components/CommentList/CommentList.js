import React from 'react';
import PropTypes from 'prop-types';
import { createComponent, } from 'react-fela';
import Comment from './Comment.js';

const propTypes = {
  comments: PropTypes.arrayOf(PropTypes.object),
  isSubComment: PropTypes.bool,
  commentsPlusRate: PropTypes.shape({
    id: PropTypes.number,
  }),
  commentsMinusRate: PropTypes.shape({
    id: PropTypes.number,
  }),
};
const defaultProps = {
  comments: [],
  isSubComment: false,
  commentsPlusRate: {},
  commentsMinusRate: {},
};

const wrapperStyle = ({ theme, }) => ({
  backgroundColor: theme.color('white'),
});

const StyledWrapper = createComponent(wrapperStyle);

function CommentList({
  comments,
  isSubComment,
  commentsPlusRate,
  commentsMinusRate,
}) {
  return (
    <StyledWrapper>
      {comments.map((comment, idx) => (
        <Comment
          key={comment.commentId}
          commentId={comment.commentId}
          author={comment.author}
          title={comment.title}
          commentText={comment.commentText}
          publishingDateForDisplay={comment.publishingDateForDisplay}
          commentNumber={idx + 1}
          subComments={comment.subComments}
          isSubComment={isSubComment}
          isEditorPick={comment.isEditorPick}
          // need to pass these down in case subComments have plus or minus rates
          commentsPlusRate={commentsPlusRate}
          commentsMinusRate={commentsMinusRate}
        />
      ))}
    </StyledWrapper>
  );
}

CommentList.propTypes = propTypes;

CommentList.defaultProps = defaultProps;

export default CommentList;
