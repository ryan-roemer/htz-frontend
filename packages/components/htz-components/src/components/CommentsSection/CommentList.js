import React from 'react';
import PropTypes from 'prop-types';
import { createComponent, } from 'react-fela';
import Comment from './Comment.js';

const propTypes = {
  /**
   * An Array of comment objects
   */
  comments: PropTypes.arrayOf(PropTypes.object),
  /**
   * An Object holding the Plus Rates of all the comments in the `CommentsElement`.
   * The Object has an id key for each comment that has at least one Plus vote.
   * The corresponding value is the rate of Plus votes the comment has.
   */
  commentsMinusRate: PropTypes.shape({
    id: PropTypes.number,
  }),
  /**
   * An Object holding the Minus Rates of all the comments in the `CommentsElement`.
   * The Object has an id key for each comment that has at least one Minus vote.
   * The corresponding value is the total number of Minus votes the comment has.
   */
  commentsPlusRate: PropTypes.shape({
    id: PropTypes.number,
  }),
  /**
   * A callback passed on to the reply `<CommentForm />`
   * @param {String} commentAuthor - the new comment author
   * @param {String} commentTextHtml - the new comment text innerHTML
   * @param {String} parentCommentId - the parent CommentId - defaults to '0' if there is no `parentCommentId`
   */
  initNewComment: PropTypes.func,
  /**
   * A callaback that initiates a vote, `<Likes />` sends up the commentId and the rate ("plus"/"minus")
   */
  initVote: PropTypes.func.isRequired,
  /** True when a CommentList renders subComments */
  isSubComment: PropTypes.bool,
  /**
   * A callback that causes the parent `<Comment />` to open its reply `<CommentForm />`
   * Replying to comments can only go one level deep so when replying to subComments,
   * we are actually replying to the parent of the subComment
   */
  openParentReplyForm: PropTypes.func,
  /** The author of the parent comment, null if it is not a subComment list */
  parentAuthor: PropTypes.string,
  /**
   * A CallBack that sends up the the commentId in order to report the Comment as abusive
   * @param {String} commentId
   */
  reportAbuse: PropTypes.func.isRequired,
  /**
   * A callback that gets the called when submitting the sign up to notification form in `<CommentSent />`
   * @param {String} - notificationEmail - The email the user entered
   */
  signUpNotification: PropTypes.func,
};
const defaultProps = {
  comments: [],
  isSubComment: false,
  commentsPlusRate: {},
  commentsMinusRate: {},
  openParentReplyForm: undefined,
  parentAuthor: null,
  initNewComment: undefined,
  signUpNotification: undefined,
};

const wrapperStyle = ({ theme, }) => ({
  backgroundColor: theme.color('white'),
});

const StyledWrapper = createComponent(wrapperStyle);

export function CommentList({
  comments,
  isSubComment,
  commentsPlusRate,
  commentsMinusRate,
  initVote,
  reportAbuse,
  parentAuthor,
  initNewComment,
  signUpNotification,
  openParentReplyForm,
}) {
  return (
    <StyledWrapper>
      {comments.map((comment, idx) => (
        <Comment
          key={comment.commentId}
          commentId={comment.commentId}
          author={comment.author}
          title={comment.title}
          commentText={comment.commentText}
          parentAuthor={parentAuthor}
          publishingDateForDisplay={comment.publishingDateForDisplay}
          commentNumber={comment.number || ''}
          subComments={comment.subComments}
          isEditorPick={comment.isEditorPick}
          // need to pass these down in case subComments have plus or minus rates
          commentsPlusRate={commentsPlusRate}
          commentsMinusRate={commentsMinusRate}
          initVote={initVote}
          reportAbuse={reportAbuse}
          initNewComment={initNewComment}
          signUpNotification={signUpNotification}
          openParentReplyForm={openParentReplyForm}
          isSubComment={isSubComment}
          isFirstSubComment={isSubComment && idx === 0}
          isLastSubComment={isSubComment && idx === comments.length - 1}
        />
      ))}
    </StyledWrapper>
  );
}

CommentList.propTypes = propTypes;

CommentList.defaultProps = defaultProps;

export default CommentList;
