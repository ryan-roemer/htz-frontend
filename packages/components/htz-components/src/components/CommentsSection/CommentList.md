`<CommentList />` is responsible for displaying a list of `<Comment />`'s, that may have.

It can be the main `<Comment />`'s list or a `subComments` list.

The `<Comment />` is responsible for indicating to the `<CommentList />` that it is a `subComment` list.

**A Comment List, with comments, subComments, editorsPick and votes**

```jsx
<div dir="rtl">
  <CommentList
    initVote={(commentId, rate) =>
      console.log(`initVote from MD commentId: ${commentId} rate: ${rate}`)
    }
    reportAbuse={commentId =>
      console.log(`report abuse from MD comment Id: ${commentId}`)
    }
    initNewComment={(commentAuthor, commentHtml, parentCommentId) =>
      console.log(
        `init Comment from CommentList.md commentAuthor: ${commentAuthor}, commentHtml: ${commentHtml} parentCommentId: ${parentCommentId}`
      )
    }
    signUpNotification={email =>
      console.log(
        `sign up comment notifications from CommentList.md email: ${email}`
      )
    }
    comments={[
      {
        author:
          "אורי ויסלאורי ויסלאורי ויסלאורי ויסלאורי ויסלאורי ויסלאורי ויסלאורי ויסלאורי ויסלאורי ויסלאורי ויסלאורי ויסלר",
        title: "התמונה בכתבה היא בסבירות גבוהה לא מבאבי יאר",
        commentText:
          "long text sdmgalkgmlaskgmaslkg lsdlk jasldk lsdkj lksajg lskdgjaslkdgjmlaskgjmlasdkgjmslkdgmsldk sdlkmg slakgmsdgasdgb;l,sad;lgb,s;adb,b fl;,b ;dflb,df;lb, dsdfagkjerngkjawn vkrj vkrejv erkjv krejgv krj vkrwej vkrjv kjrv krjv erkj vrej  v",
        publishingDateForDisplay: "08:57",
        commentId: "19.9522183",
        reviewState: "nr",
        isEditorPick: "true",
        publishingDateSortable: "20171119002030"
      },
      {
        subComments: [
          {
            author: "רן",
            title: "@הורנר@",
            commentText:
              "לגרמניה יש אובססיה להציג את ישראל של היום כגרמניה הנאצית והם לא חוסכים באמצעים. יש הרבה מאוד אנשים שמתפdfgdfsgsdfgגכדעגכעדגכעגדכעדגכעדגכעגדכעדגכעגדכעדגכעגדכעדגכעגדכעדגעדגערנסים מגדכעדגכעגדכעדגכעגדכעדגכעגדכעדגעדגערנסים משיתוגדכעדגכעגדכעדגכעגדכעדגכעגדכעדגעדגערנסים משיתוגדכעדגכעגדכעדגכעגדכעדגכעגדכעדגעדגערנסים משיתוגדכעדגכעגדכעדגכעגדכעדגכעגדכעדגעדגערנסים משיתוגדכעדגכעגדכעדגכעגדכעדגכעגדכעדגעדגערנסים משיתוגדכעדגכעגדכעדגכעגדכעדגכעגדכעדגעדגערנסים משיתוגדכעדגכעגדכעדגכעגדכעדגכעגדכעדגעדגערנסים משיתוגדכעדגכעגדכעדגכעגדכעדגכעגדכעדגעדגערנסים משיתוגדכעדגכעגדכעדגכעגדכעדגכעגדכעדגעדגערנסים משיתוגדכעדגכעגדכעדגכעגדכעדגכעגדכעדגעדגערנסים משיתוגדכעדגכעגדכעדגכעגדכעדגכעגדכעדגעדגערנסים משיתוגדכעדגכעגדכעדגכעגדכעדגכעגדכעדגעדגערנסים משיתוגדכעדגכעגדכעדגכעגדכעדגכעגדכעדגעדגערנסים משיתוגדכעדגכעגדכעדגכעגדכעדגכעגדכעדגעדגערנסים משיתוגדכעדגכעגדכעדגכעגדכעדגכעגדכעדגעדגערנסים משיתוגדכעדגכעגדכעדגכעגדכעדגכעגדכעדגעדגערנסים משיתוגדכעדגכעגדכעדגכעגדכעדגכעגדכעדגעדגערנסים משיתוגדכעדגכעגדכעדגכעגדכעדגכעגדכעדגעדגערנסים משיתוגדכעדגכעגדכעדגכעגדכעדגכעגדכעדגעדגערנסים משיתוגדכעדגכעגדכעדגכעגדכעדגכעגדכעדגעדגערנסים משיתוגדכעדגכעגדכעדגכעגדכעדגכעגדכעדגעדגערנסים משיתוגדכעדגכעגדכעדגכעגדכעדגכעגדכעדגעדגערנסים משיתוגדכעדגכעגדכעדגכעגדכעדגכעגדכעדגעדגערנסים משיתוגדכעדגכעגדכעדגכעגדכעדגכעגדכעדגעדגערנסים משיתוגדכעדגכעגדכעדגכעגדכעדגכעגדכעדגעדגערנסים משיתושיתוף הפעולה. הסיבה לאובססיה היא לא אנטישמיות קלסית זה אינטרס גרמני ברור",
            publishingDateForDisplay: "09:36",
            commentId: "19.9522362",
            reviewState: "nr",
            isEditorPick: "false",
            publishingDateSortable: "20171119002050"
          },
          {
            author: "רן",
            title: "@הורנר@",
            commentText: "testing",
            publishingDateForDisplay: "09:36",
            commentId: "19.9522999",
            reviewState: "nr",
            isEditorPick: "false",
            publishingDateSortable: "20171119002070"
          }
        ],
        author: "הורנר",
        title: "אין בזה חדש - גרמניה לא עשתה דין עם הנאצים",
        commentText:
          "גרמניה חינכה נגד השואה , בנתה אנדרטה ושילמה פיצויים . עד כאן ! היא מעולם (תחת כל ההנהגות ) לא עשתה דין ולא אסרה את הפושעים הנאצים . זכור לי ביקור של איזה פרופסור גרמני מהידלברג שמשה צימרמן הביא לאוניברסיטה להרצות לנו .. וורוד הלחיים הזה עומד מולי (ישבתי שורה ראשונה) ומבלבל לי את המוח כל כמה גרמניה אחרת ומה נעשה וטופח למדינתו על השכם ... טוב נו , באתי מוכן עם טבלת אקסל . שם / דרגה / תיאור מעלליו בשואה / עיסוקו וקורותיו ב״גרמניה האחרת״ . צימרמן ניסה להשתיק אותי אחרי השם הראשון אבל אני לא מהמושתקים ...זה לא סיפור חדש , גרמניה מגנה והגנה לכל אורך הדרך על הנאצים שלה . היא לא חשבה שיש להעניש אותם . גם כל סמולן טוב חושב שזה בסדר אכה ושדין מריך לכשות רק עם יהודים שלא מבצעים נוהל מעצר חשוד ... זה למה אני שונא בדם : 1. גרמנים . 2. סמולנים . זה למה אני אומר : יחי האביב הערבי !!! צדק קוסמי !!!",
        publishingDateForDisplay: "07:18",
        commentId: "19.9521831",
        reviewState: "nr",
        isEditorPick: "false",
        publishingDateSortable: "20171119002080"
      },
      {
        author: "נצח יהודה?",
        title: "וטובחי הנערים בכביש 443?",
        commentText: "נצחיות היודונציזם",
        publishingDateForDisplay: "06:59",
        commentId: "19.9521786",
        reviewState: "nr",
        isEditorPick: "false",
        publishingDateSortable: "20171119002090"
      }
    ]}
    commentsPlusRate={{
      "19.9522362": 0,
      "19.9521831": 4,
      "19.9521786": 11,
      "19.9522183": 122
    }}
    commentsMinusRate={{
      "19.9522362": 1,
      "19.9521831": 3,
      "19.9521786": 5
    }}
    totalHits={3}
  />
</div>
```
