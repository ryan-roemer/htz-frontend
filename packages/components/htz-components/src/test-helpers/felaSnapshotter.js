import React from 'react';
import createRenderer from '../components/StyleProvider/createRenderer';
import StyleProvider from '../components/StyleProvider/StyleProvider';
import createFelaSnapshotter from './createFelaSnapshotter';

const testRenderer = createRenderer({ isRtl: true, });
const felaSnapshotter = createFelaSnapshotter(StyleProvider, testRenderer);

export default felaSnapshotter;
