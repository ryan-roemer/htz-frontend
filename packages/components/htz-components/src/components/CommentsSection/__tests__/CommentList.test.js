/* global window */
import React from 'react';
import matchMediaPolyfill from 'mq-polyfill';
import felaSnapshotter from '../../../test-helpers/felaSnapshotter';
import CommentList from '../CommentList'; // eslint-disable-line import/no-named-as-default

// Math random used to generate random ids in TextInput,
// next row is used to produce same id everytime so tests wont fail
Math.random = jest.fn(() => 123456789);

describe('<Comment>', () => {
  describe('DOM element', () => {
    matchMediaPolyfill(window);
    window.resizeTo = resizeTo;
    window.resizeTo(1100, 768);
    it('renders correctly with required props', () => {
      const { component, styles, } = felaSnapshotter(
        <CommentList
          initVote={() => console.log('i should be a callback')}
          reportAbuse={() => console.log('i should be a callback')}
          comments={[
            {
              author:
                'אורי ויסלאורי ויסלאורי ויסלאורי ויסלאורי ויסלאורי ויסלאורי ויסלאורי ויסלאורי ויסלאורי ויסלאורי ויסלאורי ויסלר',
              title: 'התמונה בכתבה היא בסבירות גבוהה לא מבאבי יאר',
              commentText:
                'long text sdmgalkgmlaskgmaslkg lsdlk jasldk lsdkj lksajg lskdgjaslkdgjmlaskgjmlasdkgjmslkdgmsldk sdlkmg slakgmsdgasdgb;l,sad;lgb,s;adb,b fl;,b ;dflb,df;lb, dsdfagkjerngkjawn vkrj vkrejv erkjv krejgv krj vkrwej vkrjv kjrv krjv erkj vrej  v',
              publishingDateForDisplay: '08:57',
              commentId: '19.9522183',
              reviewState: 'nr',
              isEditorPick: 'true',
              publishingDateSortable: '20171119002030',
            },
          ]}
          totalHits={1}
        />,
        {
          createNodeMock: element => {
            if (element.type === 'div') {
              return { clientHeight: 24, };
            }
            return null;
          },
        }
      );
      expect(component).toMatchSnapshot();
      expect(styles).toMatchSnapshot();
    });
  });
});

function resizeTo(width, height) {
  Object.assign(this, {
    innerWidth: width,
    innerHeight: height,
    outerWidth: width,
    outerHeight: height,
  }).dispatchEvent(new this.Event('resize'));
}
