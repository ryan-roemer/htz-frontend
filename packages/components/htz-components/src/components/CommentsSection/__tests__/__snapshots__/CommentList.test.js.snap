// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Comment> DOM element renders correctly with required props 1`] = `
<div>
  <div
    className="htz-a"
  >
    <article
      className="htz-b htz-c htz-d htz-e htz-f"
    >
      <div
        className="htz-g htz-h htz-i htz-j htz-k htz-l htz-m htz-n htz-o"
      >
        <div
          className="htz-p htz-q htz-r htz-s htz-t htz-u htz-v htz-w htz-x"
        >
          <span>
            
          </span>
        </div>
        <div
          className="htz-y htz-z htz-ab htz-ac htz-ae"
        >
          <div
            className="htz-y htz-g htz-h htz-af htz-ag htz-ah htz-ai htz-m htz-n htz-aj htz-ak"
          >
            <h2
              className="htz-al htz-am htz-an htz-ao htz-ap htz-aq htz-ar htz-as htz-at htz-au htz-av htz-aw htz-ax"
              data-test="comment-author"
              onClick={[Function]}
            >
              אורי ויסלאורי ויסלאורי ויסלאורי ויסלאורי ויסלאורי ויסלאורי ויסלאורי ויסלאורי ויסלאורי ויסלאורי ויסלאורי ויסלר
            </h2>
            <span
              className="htz-q htz-ap htz-ay htz-az htz-ba htz-bb htz-bc"
            >
              08:57
            </span>
            <span
              className="htz-bd htz-ao"
            >
              | 
              בחירת העורכים
            </span>
          </div>
          <div>
            <div
              className="htz-be htz-an htz-bf"
            >
              התמונה בכתבה היא בסבירות גבוהה לא מבאבי יאר
            </div>
            <div
              className="htz-be htz-an htz-bf"
              dangerouslySetInnerHTML={
                Object {
                  "__html": "long text sdmgalkgmlaskgmaslkg lsdlk jasldk lsdkj lksajg lskdgjaslkdgjmlaskgjmlasdkgjmslkdgmsldk sdlkmg slakgmsdgasdgb;l,sad;lgb,s;adb,b fl;,b ;dflb,df;lb, dsdfagkjerngkjawn vkrj vkrejv erkjv krejgv krj vkrwej vkrjv kjrv krjv erkj vrej  v",
                }
              }
            />
          </div>
          <div
            className="htz-g htz-h htz-bg htz-ag htz-bh htz-m htz-n htz-ak"
          >
            <button
              className="htz-ag htz-bi htz-bj htz-bk htz-bl htz-bm htz-bn htz-bo htz-bp htz-bq htz-br htz-bs htz-bt htz-ao htz-bu htz-bv htz-c htz-d htz-bw htz-bx htz-by htz-bz htz-ca htz-cb htz-cc htz-cd htz-ce htz-cf htz-cg htz-ch htz-ci htz-al htz-cj htz-at htz-au htz-av htz-aw htz-ck htz-cl htz-cm htz-cn htz-co htz-cp htz-cq htz-cr htz-cs htz-ct htz-cu htz-cv htz-cw htz-cx htz-cy htz-cz htz-da htz-db htz-dc htz-dd htz-de htz-df htz-ak htz-dg htz-dh htz-di"
              id={null}
              onClick={[Function]}
              type="button"
            >
              <span
                className="htz-dj htz-dk htz-dl htz-dm htz-dn htz-do htz-dp"
              >
                הגיבו
              </span>
              <span
                className="htz-dq htz-dr htz-ds htz-dt htz-du htz-dv htz-dw htz-y htz-dx htz-dy htz-dz htz-ea htz-eb"
              />
            </button>
            <button
              className="htz-bm htz-ag htz-ec htz-ed htz-ee htz-ak"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              <svg
                className="htz-ef htz-al htz-eg htz-eh"
                height="1em"
                viewBox="0 0 256 256"
                width="1em"
              >
                  
                <path
                  d="M184 72h48v113h-48V72zM24.7 91.2c3.7 20.4 29.6 76.1 29.6 76.1 4.9 11.1 8 17.8 17.8 17.8h74c11.6 0 21-9.4 21-21V93c0-11.6-9.4-21-21-21-1 0-2-.5-2.6-1.3-20.5-27.4-24.1-41.1-30-56-1.9-4.9-6-13.8-14.8-13.8-10.9.1-24.8 13.1-25.2 28-.4 15.3 1.8 24.4 11.4 42.8 0 .1.1.1.1.2H42.9c-9.9.1-20 9.6-18.2 19.3z"
                  fill="currentColor"
                />
              </svg>
              <span
                className="htz-ei htz-ej htz-ek htz-el htz-em htz-en htz-ar htz-am htz-dt htz-eo htz-ao"
                id="like"
              >
                אהבתי
              </span>
              <span
                className="htz-q htz-ep htz-ay htz-bn htz-eq htz-au htz-er htz-es"
              >
                0
              </span>
            </button>
            <button
              className="htz-bm htz-ag htz-et htz-ed htz-cj htz-ee htz-ak"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              <svg
                className="htz-ef htz-al htz-eg htz-eh"
                height="1em"
                viewBox="0 0 256 256"
                width="1em"
              >
                  
                <path
                  d="M184 72h48v113h-48V72zM42.9 184H85c0 .1-.1.1-.1.2-9.6 18.5-11.7 27.5-11.4 42.8.4 15 14.3 28 25 28 8.8 0 12.9-8.9 14.8-13.8 5.9-14.9 9.5-28.6 30-56 .6-.8 1.6-1.3 2.6-1.3 11.6 0 21-9.4 21-21V92c0-11.6-9.4-21-21-21H72c-9.8 0-12.9 6.6-17.8 17.8 0 0-25.9 55.7-29.6 76.1-1.7 9.6 8.4 19.1 18.3 19.1z"
                  fill="currentColor"
                />
              </svg>
              <span
                className="htz-ei htz-ej htz-ek htz-el htz-em htz-en htz-ar htz-am htz-dt htz-eo htz-ao"
                id="dislike"
              >
                לא אהבתי
              </span>
              <span
                className="htz-q htz-ep htz-ay htz-bn htz-eq htz-au htz-er htz-es"
              >
                0
              </span>
            </button>
            <button
              className="htz-ag htz-bi htz-bj htz-bk htz-bl htz-bm htz-bn htz-bo htz-bp htz-bq htz-br htz-bs htz-bt htz-ao htz-bu htz-bv htz-c htz-d htz-bw htz-bx htz-by htz-bz htz-ca htz-cb htz-cc htz-cd htz-ce htz-cf htz-cg htz-ch htz-eu htz-ev htz-eq htz-au htz-er htz-es htz-ew htz-ex htz-cl htz-cm htz-ey htz-ez htz-fa htz-cq htz-cr htz-cs htz-ct htz-cu htz-fb htz-fc htz-fd htz-cy htz-cz htz-da htz-db htz-dc htz-fe htz-ff htz-fg htz-ak htz-dg htz-dh htz-di"
              id={null}
              onClick={[Function]}
              type="button"
            >
              <span
                className="htz-dj htz-dk htz-dl htz-dm htz-dn htz-do htz-dp"
              >
                דווחו כפוגעני
              </span>
              <span
                className="htz-dq htz-dr htz-ds htz-dt htz-du htz-dv htz-dw htz-y htz-dx htz-dy htz-dz htz-ea htz-eb"
              />
            </button>
          </div>
        </div>
      </div>
    </article>
  </div>
</div>
`;

exports[`<Comment> DOM element renders correctly with required props 2`] = `".htz-a{background-color:#FFF}.htz-b{background-color:#EBF2F5}.htz-c{border-bottom-width:1px}.htz-d{border-bottom-style:solid}.htz-e{border-bottom-color:#B4B4B4}.htz-f{padding-bottom:calc(2rem - 1px)}.htz-g{display:-webkit-box;display:-moz-box;display:-ms-flexbox;display:-webkit-flex;display:flex}.htz-h{flex-direction:row}.htz-i{justify-content:flex-end}.htz-j{padding-right:1rem}.htz-k{padding-left:1rem}.htz-l{padding-top:2rem}.htz-m{-webkit-box-orient:horizontal}.htz-n{-webkit-box-direction:normal}.htz-o{-webkit-box-pack:left}.htz-p{width:8.5rem}.htz-q{color:#787878}.htz-r{flex-shrink:0}.htz-s{flex-grow:0}.htz-x{-webkit-box-flex:0}.htz-y{width:100%}.htz-z{flex-shrink:1}.htz-ab{flex-grow:1}.htz-ac{min-width:0}.htz-ae{-webkit-box-flex:1}.htz-af{justify-content:flex-start}.htz-ag{align-items:center}.htz-aj{-webkit-box-pack:right}.htz-ak{-webkit-box-align:center}.htz-al{color:#0B7EB5}.htz-am{padding:0}.htz-an{margin:0}.htz-ao{white-space:nowrap}.htz-ap{margin-left:1rem}.htz-aq{max-width:50%}.htz-ar{overflow:hidden}.htz-as{text-overflow:ellipsis}.htz-ax:hover{cursor:pointer}.htz-ay{margin-right:1rem}.htz-bd{color:#169FD1}.htz-be{color:#2D2D2D}.htz-bf{word-break:break-word}.htz-bg{align-content:flex-start}.htz-bh{margin-top:2rem}.htz-bi{-webkit-appearance:none}.htz-bj{-moz-appearance:none}.htz-bk{appearance:none}.htz-bl{cursor:pointer}.htz-bm{display:-webkit-inline-box;display:-moz-inline-box;display:-ms-inline-flexbox;display:-webkit-inline-flex;display:inline-flex}.htz-bn{font-weight:bold}.htz-bo{font-size:inherit}.htz-bp{justify-content:center}.htz-bq{position:relative}.htz-br{transform:scale(1)}.htz-bs{transition:all 0.16666666666666666s cubic-bezier(0, 0, .2, 0) , transform .3s cubic-bezier(.55, 0, .1, 1) .3s, backgroundColor .25s cubic-bezier(0, 0, .2, 0)}.htz-bt{text-decoration:none}.htz-bu::moz-focus-inner{border:0}.htz-bv::moz-focus-inner{padding:0}.htz-bw{padding-bottom:calc(0.5rem - 1px)}.htz-bx{border-top-width:1px}.htz-by{border-top-style:solid}.htz-bz{padding-top:calc(0.5rem - 1px)}.htz-ca{border-right-width:1px}.htz-cb{border-right-style:solid}.htz-cc{padding-right:2rem}.htz-cd{border-left-width:1px}.htz-ce{border-left-style:solid}.htz-cf{padding-left:2rem}.htz-cg{border-radius:0}.htz-ch{background-color:transparent}.htz-ci{border-color:#0B7EB5}.htz-cj{margin-left:2rem}.htz-ck:visited{color:#0B7EB5}.htz-cl:hover{text-decoration:none}.htz-cm:hover{transition:all 0.16666666666666666s cubic-bezier(.55, 0, .1, 1) , transform .3s cubic-bezier(.55, 0, .1, 1) .3s, backgroundColor .25s cubic-bezier(.55, 0, .1, 1)}.htz-cn:hover{background-color:#EBF2F5}.htz-co:hover{border-color:#006B96}.htz-cp:hover{color:#006B96}.htz-cq:hover{-webkit-transition:all 0.16666666666666666s cubic-bezier(.55, 0, .1, 1) , transform .3s cubic-bezier(.55, 0, .1, 1) .3s, backgroundColor .25s cubic-bezier(.55, 0, .1, 1)}.htz-cr:hover{-moz-transition:all 0.16666666666666666s cubic-bezier(.55, 0, .1, 1) , transform .3s cubic-bezier(.55, 0, .1, 1) .3s, backgroundColor .25s cubic-bezier(.55, 0, .1, 1)}.htz-cs:focus{outline:none}.htz-ct:focus{text-decoration:none}.htz-cu:focus{transition:all 0.16666666666666666s cubic-bezier(.55, 0, .1, 1) , transform .3s cubic-bezier(.55, 0, .1, 1) .3s, backgroundColor .25s cubic-bezier(.55, 0, .1, 1)}.htz-cv:focus{background-color:#0B7EB5}.htz-cw:focus{border-color:#0B7EB5}.htz-cx:focus{color:#FFF}.htz-cy:focus{-webkit-transition:all 0.16666666666666666s cubic-bezier(.55, 0, .1, 1) , transform .3s cubic-bezier(.55, 0, .1, 1) .3s, backgroundColor .25s cubic-bezier(.55, 0, .1, 1)}.htz-cz:focus{-moz-transition:all 0.16666666666666666s cubic-bezier(.55, 0, .1, 1) , transform .3s cubic-bezier(.55, 0, .1, 1) .3s, backgroundColor .25s cubic-bezier(.55, 0, .1, 1)}.htz-da:active{outline:none}.htz-db:active{text-decoration:none}.htz-dc:active{transition-duration:0s}.htz-dd:active{background-color:#EBF2F5}.htz-de:active{border-color:#0B7EB5}.htz-df:active{color:#006B96}.htz-dg{-webkit-box-pack:center}.htz-dh{-webkit-transition:all 0.16666666666666666s cubic-bezier(0, 0, .2, 0) , transform .3s cubic-bezier(.55, 0, .1, 1) .3s, backgroundColor .25s cubic-bezier(0, 0, .2, 0)}.htz-di{-moz-transition:all 0.16666666666666666s cubic-bezier(0, 0, .2, 0) , transform .3s cubic-bezier(.55, 0, .1, 1) .3s, backgroundColor .25s cubic-bezier(0, 0, .2, 0)}.htz-dj{opacity:1}.htz-dk{transition-property:opacity}.htz-dl{transition-duration:0.16666666666666666s}.htz-dm{transition-timing-function:cubic-bezier(.55, 0, .1, 1)}.htz-dn{transition-delay:.3s}.htz-do{-webkit-transition-property:opacity}.htz-dp{-moz-transition-property:opacity}.htz-dq{background-color:rgba(200,200,200, 0.3)}.htz-dr{height:100%}.htz-ds{right:0}.htz-dt{position:absolute}.htz-du{top:0}.htz-dv{transform:scaleX(0)}.htz-dw{transform-origin:right}.htz-dx{animation-direction:alternate}.htz-dy{animation-iteration-count:infinite}.htz-dz{animation-duration:1s}.htz-ea{animation-delay:0.75s}.htz-eb{animation-timing-function:cubic-bezier(.55, 0, .1, 1)}.htz-ec{margin-right:auto}.htz-ed:disabled{cursor:default}.htz-ef{vertical-align:-0.15em}.htz-eg{fill:#FFF}.htz-eh{font-size:3.5rem}.htz-ei{border:0}.htz-ej{clip:rect(0 0 0 0)}.htz-ek{-webkit-clip-path:inset(50%)}.htz-el{clip-path:inset(50%)}.htz-em{height:1px}.htz-en{margin:-1px}.htz-eo{width:1px}.htz-ep{vertical-align:center}.htz-et{margin-right:2rem}.htz-eu{border-color:transparent}.htz-ev{color:#A8001C}.htz-ex:visited{color:#A8001C}.htz-ey:hover{background-color:#FEE}.htz-ez:hover{border-color:transparent}.htz-fa:hover{color:#8A021B}.htz-fb:focus{background-color:#A8001C}.htz-fc:focus{border-color:transparent}.htz-fd:focus{color:#fff}.htz-fe:active{background-color:#fff !important}.htz-ff:active{border-color:transparent}.htz-fg:active{color:#A8001C}@media (max-width: 79.9375em){.htz-t{font-size:4rem}.htz-u{line-height:1.25em}.htz-at{font-size:2.6666666666666665rem}.htz-au{line-height:1.5em}.htz-az{font-size:2.3333333333333335rem}.htz-ba{line-height:1.2857142857142858em}.htz-eq{font-size:2rem}}@media (min-width: 80em){.htz-v{font-size:3.857142857142857rem}.htz-w{line-height:1.2962962962962963em}.htz-av{font-size:2.5714285714285716rem}.htz-aw{line-height:1.5555555555555556em}.htz-bb{font-size:2.2857142857142856rem}.htz-bc{line-height:1.3125em}.htz-er{font-size:2rem}.htz-es{line-height:1.5em}}@media (max-width: 37.4375em){.htz-ah{flex-wrap:wrap}.htz-ai{-webkit-box-lines:multiple}.htz-ee{display:none}.htz-ew{margin-right:auto}}"`;
